content/reactions/corrodes.ts

```diff
@@ -85,6 +86,99 @@ export const applications: Reactions = {
     };
   },
 
+  // dissolve a level of clay
+  [Descriptor.Clay]: (args: ReactionExtendedArgs) => {
+
+    const corrodesLevel = getInteractionLevel(args.sourceItem, Interaction.Corrodes);
+
+    const sourceItem = args.sourceItem;
+    const targetItem = args.targetItem;
+
+    if (corrodesLevel <= 0)
+      return zeroFail(args);
+
+    decreaseInteractionLevel(args.sourceItem, Interaction.Corrodes, 1);
+    decreaseDescriptorLevelForPart(args.targetPart, Descriptor.Clay, 1);
+
+    if (getDescriptorLevelFromPart(args.targetPart, Descriptor.Clay) <= 0) {
+      return {
+        message: 'Completely dissolved the clay.',
+        success: true,
+        newSource: sourceItem,
+        newTarget: undefined
+      };
+    }
+
+    return {
+      message: 'Dissolved some of the clay.',
+      success: true,
+      newSource: sourceItem,
+      newTarget: targetItem
+    };
+  },
+
+  // dissolve a level of dirt
+  [Descriptor.Dirt]: (args: ReactionExtendedArgs) => {
+
+    const corrodesLevel = getInteractionLevel(args.sourceItem, Interaction.Corrodes);
+
+    const sourceItem = args.sourceItem;
+    const targetItem = args.targetItem;
+
+    if (corrodesLevel <= 0)
+      return zeroFail(args);
+
+    decreaseInteractionLevel(args.sourceItem, Interaction.Corrodes, 1);
+    decreaseDescriptorLevelForPart(args.targetPart, Descriptor.Dirt, 1);
```

Comment:
this can be `const newDirtLevel = decreaseDesc...` - the level changing functions should always return their value